File: php/003/main.php
<?php
/*
 * ProjectEuler 003
 * PHP / main.php
 * copyright (c) 2015 Susisu
 *
 * PHP 5.4.30
 * `php main.php`
 */

function main() {
    $n = 600851475143;
    $p = 2;
    while ($n > 1) {
        if ($n % $p === 0) {
            $n /= $p;
        }
        else {
            $p++;
        }
    }
    echo "$p\n";
}

main();

File: rust/003/main.rs
/*
    ProjectEuler 003
    Rust / main.rs
    copyright (c) 2015 Susisu

    Rust 1.3.0
    `rustc main.rs -o bin/main`
*/

fn main() {
    let mut n: i64 = 600851475143;
    let mut d: i64 = 2;
    while n > 1 {
        if n % d == 0 {
            n /= d;
        }
        else {
            d += 1;
        }
    }
    println!("{}", d);
}
